### PR TITLE
[List] Add RTL Arabic text to list example

### LINKED ISSUE #5498: [List] Add label introspection with subsequent textAlignment setting to List example
## Feature Request/Suggestion

It would be nice to have some code in our List example showing how one might determine the correct textAlignment for a label based off the language of its text.

### PR DESCRIPTION
**Pull request by andrewoverton**
## The problem:

There wasn't any RTL text in our List example!

## The solution:

I added some RTL text to our List example.

Closes #5498

### Before:
simulator screen shot - iphone x - 2018-10-23 at 11 28 37

### After:
simulator screen shot - iphone x - 2018-10-23 at 11 28 02

### CODE REVIEW

File: components/List/examples/MDCSelfSizingStereoCellExample.m [hunk 1]
```diff
@@ -122,21 +126,50 @@ - (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView
...
-  for (NSInteger i = 0; i < numberOfWords; i++) {
-    NSInteger lengthOfWord = 0 + arc4random() % (10 - 0);
-    NSMutableArray *letterArray = [[NSMutableArray alloc] initWithCapacity:lengthOfWord];
-    for (NSInteger j = 0; j < lengthOfWord; j++) {
-      int asciiCode = 97 + arc4random() % (122 - 97);
-      NSString *characterString = [NSString stringWithFormat:@"%c", asciiCode];
-      [letterArray addObject:characterString];
+- (NSTextAlignment)textAlignmentForText:(NSString *)text {
+  if (text.length > 0) {
+    if (text.mdf_calculatedLanguageDirection == NSLocaleLanguageDirectionLeftToRight) {
```
Comment (ianegordon) [inline components/List/examples/MDCSelfSizingStereoCellExample.m:131]: There may be some performance impact from using mdf_calculatedLanguageDirection in cellForIndexPath, but if everything looks okay, LGTM.

**Review by ianegordon — APPROVED** [review]
Please consider comment before merge.

File: components/List/examples/MDCSelfSizingStereoCellExample.m (hunk 1 above)
Comment (romoore) [reply components/List/examples/MDCSelfSizingStereoCellExample.m:131]: Was the performance impact measured in scrolling? Can we still hit 60 fps on an iPhone 5c/iPhone 4S?

Comment (romoore) [reply components/List/examples/MDCSelfSizingStereoCellExample.m:131]: I did a quick-and-dirty test on an iPhone 5c/iOS 10.3.3 and found that for 14 seconds worth of casual scrolling (over a 30 second period, depending on trace), it added about 116 ms worth of overhead (~0.8% increase).

mdc-list-example-mdf-overhead

Comment (andrewoverton) [reply components/List/examples/MDCSelfSizingStereoCellExample.m:131]: Interesting! Are we okay with this?

Comment (romoore) [reply components/List/examples/MDCSelfSizingStereoCellExample.m:131]: I dropped the performance to about half by caching the result in #5506

**Pull request merged**